raiden/tests/integration/network/transport/test_web_rtc.py

```diff
@@ -0,0 +1,87 @@
+import gevent
+import pytest
+from gevent.timeout import Timeout
+
+from raiden.constants import EMPTY_SIGNATURE
+from raiden.messages.synchronization import Processed
+from raiden.network.transport.matrix.transport import MessagesQueue
+from raiden.tests.utils import factories
+from raiden.tests.utils.mocks import MockRaidenService
+from raiden.transfer import views
+from raiden.transfer.identifiers import QueueIdentifier
+from raiden.utils.formatting import to_checksum_address
+from raiden.utils.typing import MessageID
+
+TIMEOUT_MESSAGE_RECEIVE = 15
+
+
+class MessageHandler:
+    def __init__(self, bag):
+        self.bag = bag
+
+    def on_messages(self, _, messages):
+        self.bag.update(messages)
+
+
+@pytest.mark.parametrize("matrix_server_count", [1])
+@pytest.mark.parametrize("number_of_transports", [2])
+def test_matrix_message_sync(matrix_transports):
+
+    transport0, transport1 = matrix_transports
+
+    transport0_messages = set()
+    transport1_messages = set()
+
+    transport0_message_handler = MessageHandler(transport0_messages)
+    transport1_message_handler = MessageHandler(transport1_messages)
+
+    raiden_service0 = MockRaidenService(transport0_message_handler)
+    raiden_service1 = MockRaidenService(transport1_message_handler)
+
+    print(f"transport0: {to_checksum_address(raiden_service0.address)}")
+    print(f"transport1: {to_checksum_address(raiden_service1.address)}")
+
+    transport0.start(raiden_service0, [], None)
+    transport1.start(raiden_service1, [], None)
+
+    transport0.immediate_health_check_for(transport1._raiden_service.address)
+    transport1.immediate_health_check_for(transport0._raiden_service.address)
+
+    while raiden_service1.address not in transport0._web_rtc_manager.address_to_rtc_partners:
+        gevent.sleep(1)
+    while raiden_service0.address not in transport1._web_rtc_manager.address_to_rtc_partners:
+        gevent.sleep(1)
+
+    queue_identifier = QueueIdentifier(
+        recipient=transport1._raiden_service.address,
+        canonical_identifier=factories.UNIT_CANONICAL_ID,
+    )
+
+    raiden0_queues = views.get_all_messagequeues(views.state_from_raiden(raiden_service0))
+    raiden0_queues[queue_identifier] = []
+
+    for i in range(5):
+        message = Processed(message_identifier=MessageID(i), signature=EMPTY_SIGNATURE)
```

Comment:
How can we be sure that this is send with WebRTC?

Reply:
I rewrote the test, please check again